contributors/design-proposals/service-policy.md

```diff
@@ -0,0 +1,68 @@
+# Service Policy Proposal
+
+## Abstract
+
+ServicePolicy allows cluster administrators to control the creation and validation of a service.
+The intent of ServicePolicy is to manage the services.
+
+## Motivation
+
+As we know, field level control over allowed fields in services does not exist yet. we can not limit
+the ip range for different users. and if we don't want people use the node port directly, we can not
```

Comment:
Could you add some detail here into the use-cases you're considering for limiting the IP range and nodePort range for different users?

NodePorts are in my mind primarily intended to be a resource where you don't _care_ about the specific port that has been allocated since they will usually be behind a loadbalancer of some sort anyway.

ClusterIPs are a bit different, but similarly it's best not to make too many assumptions about the range or value you might be assigned else you break application portability across clusters.

Reply:
> Could you add some detail here into the use-cases you're considering for limiting the IP range and nodePort range for different users?
NodePorts are in my mind primarily intended to be a resource where you don't care about the specific port that has been allocated since they will usually be behind a loadbalancer of some sort anyway.
ClusterIPs are a bit different, but similarly it's best not to make too many assumptions about the range or value you might be assigned else you break application portability across clusters.

Since loadbanlancer is suggested to use, I think we could limit the use of NodePorts (e.g. disable  nodeport).
If we can cut clusterIPs into slices,  may be can reduce the collisions?

Reply:
We can already limit nodeports with quota.

Reply:
Reducing collisions of service ips doesn’t sound very useful.  Explain more, please.